proto/backup.proto

```diff
@@ -214,3 +214,32 @@ message BackupResponse {
 service Backup {
     rpc backup(BackupRequest) returns (stream BackupResponse) {}
 }
+
+// ExternalStorage is a service for using a cloud backend from StorageBackend to store files.
+// This can be used to backup and restore SST files.
+service ExternalStorage {
+    // Restore to a file
+    rpc restore(ExternalStorageRestoreRequest) returns (ExternalStorageRestoreResponse) {}
```

Comment:
Why not name it `read`?

Reply:
ExternalStorage in TiKV has a "read" method that creates a read stream from S3. The restore method goes a step further and saves it to a file.

The direction that would make more sense to me would be to instead call this write API (which takes a file, not a stream) something like "save". More descriptive would be "save file to S3"